dts: bindings: stepper: use hyphens instead of underscore

jilaypandya opened the pull request:
This commit replaces hyphens with underscore in order to comply with devicetree standards

jilaypandya committed:
dts: bindings: stepper: use hyphens instead of underscore

This commit replaces hyphens with underscore in order to comply with
devicetree standards

Signed-off-by: Jilay Pandya <jilay.pandya@outlook.com>

rruuaanng commented:
That's great! But it seems there are no changes to the DTS files. Hopefully, we didn't miss any DTS files.

kartben reviewed (changes requested):
needs migration guide entry (unless they were added post 4.0, maybe?)

rruuaanng commented:
Perhaps the proposed changes will be implemented in the 4.1 release. Adding a migration for version 4.1 might be more appropriate, as these property names are still present in the 4.0 release.
(I think)

jilaypandya committed:
docs: migration_guide_4_1: migration guide entry for renamed properties

add migration guide for following renamed variables
 - ``en_spreadcycle`` to ``en-spreadcycle``
- ``i_scale_analog`` to ``i-scale-analog``
- ``index_optw``to ``index-otpw```
- ``ìndex_step`` to ``index-step``
- ``internal_rsense`` to ``internal-rsense``
- ``lock_gconf`` to ``lock-gconf``
- ``mstep_reg_select`` to ``mstep-reg-select``
- ``pdn_disable`` to ``pdn-disable``
- ``poscmp_enable`` to ``poscmp-enable``
- ``test_mode`` to ``test-mode``

Signed-off-by: Jilay Pandya <jilay.pandya@outlook.com>

rruuaanng commented on doc/releases/migration-guide-4.1.rst:
```diff
@@ -279,6 +279,17 @@ Stepper
     :kconfig:option:`STEPPER_ADI_TMC5041_RAMP_GEN` option.
   * To control the velocity for :dtcompatible:`adi,tmc5041` stepper driver, use
     :c:func:`tmc5041_stepper_set_max_velocity` or :c:func:`tmc5041_stepper_set_ramp`.
+  * Renamed following DeviceTree properties:
```
Perhaps listing them individually would be better, as there are too many of them, and this approach would align better with the existing style of the migration entry. 

E.g.
```
* Renamed the devicetree property ``en_spreadcycle`` to ``en-spreadcycle``.
* ...
```

jilaypandya commented:
> needs migration guide entry (unless they were added post 4.0, maybe?)

Done
